IdentityCore/src/cache/MSIDTokenCache.h

```diff
@@ -25,33 +25,86 @@
+                            error:(NSError **)error;
 
-- (MSIDToken *)getMsalAT:(NSString *)userId
-                  scopes:(NSSet<NSString *> *)scopes
-                clientId:(NSString *)clientId
-                   error:(NSError **)error;
+- (MSIDToken *)getFRTforUser:(MSIDUser *)user
+                   authority:(NSURL *)authority
+                    familyId:(NSString *)familyId
+                     context:(id<MSIDRequestContext>)context
+                       error:(NSError **)error;
 
-- (MSIDToken *)getFRT:(NSString *)userId
-             familyId:(NSString *)familyId
-                error:(NSError **)error;
+- (MSIDToken *)getAdfsUserTokenForAuthority:(NSURL *)authority
+                                   resource:(NSString *)resource
+                                   clientId:(NSString *)clientId
+                                    context:(id<MSIDRequestContext>)context
+                                      error:(NSError **)error;
+
+- (BOOL)getMsalATwithAuthority:(NSURL *)authority
+                      clientId:(NSString *)clientId
+                        scopes:(NSOrderedSet<NSString *> *)scopes
+                          user:(MSIDUser *)user
+                   accessToken:(MSIDToken **)outAccessToken
+                authorityFound:(NSURL **)outAuthorityFound
+                       context:(id<MSIDRequestContext>)context
+                         error:(NSError **)error;
+
+- (MSIDToken *)getRTforUser:(MSIDUser *)user
+                  authority:(NSURL *)authority
+                   clientId:(NSString *)clientId
+                    context:(id<MSIDRequestContext>)context
+                      error:(NSError **)error;
 
-- (MSIDToken *)getAdfsUserToken:(NSString *)resource
-                       clientId:(NSString *)clientId
-                          error:(NSError **)error;
+- (NSArray<MSIDToken *> *)getAllRTsForClientId:(NSString *)clientId
+                                       context:(id<MSIDRequestContext>)context
+                                         error:(NSError **)error;
 
-- (NSArray<MSIDToken *> *)getRTs:(NSString *)clientId
-                           error:(NSError **)error;
+- (BOOL)saveAdalAT:(MSIDToken *)adalAT
+          clientId:(NSString *)clientId
+              user:(MSIDUser *)user
+          resource:(NSString *)resource
+           context:(id<MSIDRequestContext>)context
+             error:(NSError **)error;
 
+- (BOOL)saveMsalAT:(MSIDToken *)msalAT
+         authority:(NSURL *)authority
+          clientId:(NSString *)clientId
+              user:(MSIDUser *)user
+            scopes:(NSOrderedSet<NSString *> *)scopes
+           context:(id<MSIDRequestContext>)context
+             error:(NSError **)error;
+
+- (BOOL)saveRT:(MSIDToken *)rt
+          user:(MSIDUser *)user
+     authority:(NSURL *)authority
+      clientId:(NSString *)clientId
+       context:(id<MSIDRequestContext>)context
+         error:(NSError **)error;
+
+- (BOOL)saveAdfsToken:(MSIDToken *)adfsToken
```

Comment:
Should it save a different type of token?

Reply:
I believe yes, a third type might be in need here